docs/execution.md

```diff
@@ -0,0 +1,208 @@
+Possible optimizations include caching, checkpointing, incremental updates, etc... but are beyond the scope of this document.
+
+**Whenever some blocks become final**, the CSS notifies the SCE of all new final blocks of using the `BlocksFinal(BlockHashMap<Block>)` message.
+
+```rust
+BlockcliqueChanged {
+    blockclique: BlockHashMap<Block>,
+    finalized: BlockHashMap<Block>
+}
+```
+
+The SCE locally maintains a Final SCE ledger and a Candidate SCE ledger.
+Addresses in the CSS ledger are also part of the SCE ledgers.
+Smart-contract specific addresses are also part of the SCE ledger (but not CSS ledgers).
+
+When an SCE block execution requests a coin transfer form the SCE ledger to the CSS ledger, a `TransferToConsensus{target_addr: Address, origin_slot: Slot, amount: Amount}` message is sent from the SCE to the CSS.
+`origin_slot` is the slot of the block causing the `TransferToConsensus`.
+
+When the SCE receives a `BlockCliqueChanged` message:
+* the SCE reset its Candidate SCE ledger to the Final SCE ledger
+* the SCE sorts all blockclique blocks by increasing slot
+* for each block B in that order, the SCE processes the ExecuteSC operations in the block in the order they appear in the block. For each such operation `op`:
+  * credit the block creator in the SCE ledger with `op.max_gas * op.gas_price`
+  * execute the smart contract bytecode (rollback and ignore in case of failure, but to not reimburse gas fees)
+    
+When the SCE receives a `BlocksFinal` message:
+* this message notifies the SCE about new CSS-final blocks
+* a slot is SCE-final if its immediate predecessor slot contains a block that is CSS-final or is empty but followed in its own thread by a CSS-final block. This ensures that no new block executions can be inserted in-between existing final block executions in the SCE
+
+When a block B becomes SCE-final:
+* update the Final SCE ledger by VM-executing the newly SCE-final block on top of the current Final SCE ledger 
+* if the block B requires transferring coins from the SCE ledger to the CSS ledeger, send a TransferToConsensus to the CSS:
+  * fields:
+    * target_addr = the emitter of the operation that caused the TransferToConsensus
+    * origin_slot = the slot of the SCE-final block causing the TransferToConsensus
+    * amount = amount of coins to transfer
+  * this will cause the spending of "amount" coins on the SCE side
+
+When the CSS receives a TransferToConsensus message:
+* it registers that `target_addr` needs to be credited of `amount` coins in the CSS ledger at slot `target_slot = origin_slot + 1 period`
+* the credit is performed at the beginning of the first block that comes at `target_slot` (or later in case of misses)
+
+
+### Structure of the SCE
+
+
+```mermaid
+flowchart TB
+    Address-->Bytecode
+    subgraph SCE Ledger Data
+      Address-->Balance
+      Address-->DB
+    end
+    subgraph Program
+      Bytecode
+    end
```

Comment:
I don't understand why that part of the graph is outside of the ledger data ?

Reply:
looks like github does not render mermaid ?

https://github.com/massalabs/massa/blob/ebeca6cf9c515ed62b88a143f769c883d8ebfdb8/docs/execution.md

Reply:
It's not supported but you can install a plug in https://addons.mozilla.org/fr/firefox/addon/github-mermaid/

Reply:
that's unfortunate...

Reply:
> It's not supported but you can install a plug in https://addons.mozilla.org/fr/firefox/addon/github-mermaid/

This quote resolved the conversation